src/Input/Select/Select.test.tsx

```diff
@@ -360,3 +360,196 @@ it('up and down arrow keys can be used to navigate the options menu', () => {
   fireEvent.keyDown(selectInput, arrowUpEvent);
   expect(firstOption).toHaveClass('active');
 });
+
+describe('<Select/> on click outside', () => {
+  it('should lose focus', async () => {
+    const Component = () => (
+      <React.Fragment>
+        <Select label="ClickOutside">
+          <Select.Option value="ClickOutside">ClickOutside</Select.Option>
+        </Select>
+        <div>outside</div>
+      </React.Fragment>
+    );
+
+    const { queryByTestId, queryByRole, queryByText } = render(<Component />);
+    const selectInput = queryByRole('combobox') as HTMLInputElement;
+    const selectList = queryByTestId('listbox');
+    const outside = queryByText('outside');
+
+    fireEvent.focus(selectInput);
+    expect(selectList.hasAttribute('open')).toEqual(true);
+    fireEvent.click(outside);
+    expect(selectList.hasAttribute('open')).toEqual(false);
+    expect(selectInput).not.toHaveFocus();
+  });
+
+  it('should remove click outside listener when unmount', () => {
+    const addEventListenerSpy = jest.spyOn(document, 'addEventListener');
+    const removeEventListenerSpy = jest.spyOn(document, 'removeEventListener');
+
+    const Component = () => (
+      <React.Fragment>
+        <Select label="ClickOutside">
+          <Select.Option value="ClickOutside">ClickOutside</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { unmount } = render(<Component />);
+    unmount();
+    expect(removeEventListenerSpy.mock.calls[0][1]).toEqual(
+      addEventListenerSpy.mock.calls[0][1]
+    );
+    addEventListenerSpy.mockRestore();
+    removeEventListenerSpy.mockRestore();
+  });
+});
+
+describe('<Select/> onBlur onFocus', () => {
+  it('should call onBlur and onFocus', () => {
+    const onBlurSpy = jest.fn();
+    const onFocusSpy = jest.fn();
+    const Component = () => (
+      <React.Fragment>
+        <Select label="ClickOutside" onBlur={onBlurSpy} onFocus={onFocusSpy}>
+          <Select.Option value="ClickOutside">ClickOutside</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { queryByRole, asFragment } = render(<Component />);
+    const selectInput = queryByRole('combobox') as HTMLInputElement;
+
+    fireEvent.focus(selectInput);
+    expect(onFocusSpy).toHaveBeenCalledTimes(1);
+    // use snapshot to validate isFocus
+    expect(asFragment()).toMatchSnapshot();
```

Comment:
May I know the reason why you use `toMatchSnapshot` rather than `toHaveFocus`?

Reply:
because we can validate it from the snapshot.